Add internal docs on releasing from a support branch (#1507)
alphagov/govuk-prototype-kit: internal_docs/releasing-from-a-support-branch.md

# Releasing from a support branch

This document is for Prototype team developers who need to publish a support branch of the GOV.UK Prototype Kit. For example, you might need to release a fix as part of a:

- patch release, after the team has started to merge changes for a new feature release into the `main` branch - for example, a 13.14.x release once we've started merging changes for 13.15.0
- release, after the team has started to merge changes for a new breaking release into the `main` branch - for example, a 12.x.x release once we've started merging changes for 13.0.0
- release for a previous major version - for example, a 12.x.x release after we've released 13.0.0

> If you want to publish the `main` branch for the prototype kit, [follow the steps in releasing a new version of the prototype kit](/internal_docs/releasing.md).

Reviewer: ```suggestion
If you want to publish the `main` branch for the Prototype Kit, [follow the steps in releasing a new version of the prototype kit](/internal_docs/releasing.md).
```

If the `main` branch only has a few unreleasable changes, you can temporarily revert these changes.

1. Revert the unreleasable changes on the `main` branch.

> 2. Publish the prototype kit.

Reviewer: ```suggestion
2. Publish the Prototype Kit.
```

3. Add the reverted changes back into the `main` branch.

However, this approach has risks. For example, it creates a messy commit history on the `main` branch.

## Before you release

> 1. Draft release notes in a Google Doc

Reviewer: ```suggestion
1. Draft release notes in a Google Doc.
```

> ## Release a new version of the prototype kit from the support branch

Reviewer: ```suggestion
## Release a new version of the Prototype Kit from the support branch
```

9. Commit your changeds and open a new pull request, with `support/<MAJOR VERSION NUMBER>.x` as the base branch to merge into. Copy the relevant Changelog section into the description.

10. Once a developer approves the pull request, merge it into `support/<MAJOR VERSION NUMBER>.x`.

### Publish the release to GitHub

1. [Draft a new release on GitHub](https://github.com/alphagov/govuk-prototype-kit/releases) with the target `support/<MAJOR VERSION NUMBER>.x`.

2. In Tag version and Release title, put v[version-number], for example `v12.2.0`.

3. In the description, paste the relevant section from the release notes in the Google Doc.

4. Check out the `support/<MAJOR VERSION NUMBER>.x` branch and pull the latest changes.

5. Run `node scripts/create-release-archive`, which will generate a ZIP in the root of this project.

6. Attach the generated ZIP to the release.

7. Click 'Publish release'.

## After you publish the new release

1. Let the community know about the release.

Write a brief summary with highlights from the release then send it to the following slack channels:

- X-GOV #govuk-design-system
- X-GOV #prototype-kit
- GDS #govuk-design-system

> Make sure to send a link to the install page rather than the GitHub release page: https://govuk-prototype-kit.herokuapp.com/docs/install.

Reviewer: ```suggestion
Include a link to the install page: https://govuk-prototype-kit.herokuapp.com/docs/install.

Include a link to the and the GitHub release page if there are actions for users that are not covered in the release notes.
```

> 2. On the [GOV.UK Prototype Team Sprintboard](https://github.com/orgs/alphagov/projects/15):

Reviewer: ```suggestion
2. On the [GOV.UK Prototype team Sprintboard](https://github.com/orgs/alphagov/projects/15):
```
